coda/coda_validate/tests/test_utils.py

```diff
@@ -64,9 +65,11 @@ def test_xmlToValidateObject(validate_feed):
     assert validate_xml.server == validate.server
 
 
-@pytest.mark.xfail(reason='Exception will never be raised directly from the function.')
-def test_xmlToValidateObject_raises_exception():
-    assert 0
+@mock.patch('coda_validate.views.etree.XML', return_value=None)
+def test_xmlToValidateObject_raises_exception(mock_xml):
+    with pytest.raises(Exception) as e:
```

Comment:
We should probably catch a specific exception here and in the other similar test.

Reply:
So I didn't realize at the time that you _were_ catching as specific an exception as you could. Odd that we'd raise a plain Exception. So, while my intention wasn't to change the called code, maybe this is a good thing? I can't image there being any unforeseen consequences of raising (and catching) a more specific exception, as anything that was looking for the base Exception class should still catch this more specific one, but still, thoughts @ldko, @madhulika95b? Is this a safe change?

Reply:
well.. I tried raising `ParseError`, `ValueError` exceptions in the source code and it didn't raise any exception but `ValidationError` raised the exception. And mostly these exceptions are raised when dealing with xml data parsing. So I thought this would be safe. Do you think I should add any more exceptions here?

Reply:
The specific exception raised doesn't bother me, I just hadn't made this comment with the expectation that it would affect the code being tested. I originally thought there _already_ was a more specific subclass of Exception being raised, and that we should catch whatever that was. But I see that's not the case. Had I known that at the time, I wouldn't have commented at all, but since the change has been made, I'm just wanting to make sure that it's a _safe_ change, which I believe it is, but I wanted to get other opinions. I tend to be very cautious about changing the code under test rather than _just_ changing the test itself.

Reply:
Yes, what you said is totally true. 'cos catching specific exceptions is a good habit generally, I thought it's worth changing it. Let's see what @ldko has to say abt it.

Reply:
> well.. I tried raising ParseError, ValueError exceptions in the source code and it didn't raise any exception but ValidationError raised the exception. And mostly these exceptions are raised when dealing with xml data parsing. So I thought this would be safe. Do you think I should add any more exceptions here?

The code that has been changed in `codavalidate/views.py` is to explicitly raise a certain type of exception, so I am not sure why that wouldn't raise an exception. I think it is ok to raise a more specific exception, but I don't think [`ValidationError`](https://docs.djangoproject.com/en/1.11/ref/exceptions/#validationerror), a Django exception  "raised when data fails form or model field validation" is necessarily the right choice. I think `ValueError` makes a little more sense, which is why I am wondering about the comment saying it didn't raise any exception.

Reply:
Umm...I tried changing the exception now, and it worked. But I got error when I tried using ValueError before. I guess my test used cached test code which didn't match with the raised exception. But I can use `ValueError` in raising exception.